app/src/main/java/net/bonysoft/magicmirror/facerecognition/KeyboardFaceSource.java

```diff
@@ -0,0 +1,57 @@
+package net.bonysoft.magicmirror.facerecognition;
+
+public class KeyboardFaceSource implements FaceReactionSource {
+
+    private static final FaceExpression NEUTRAL_EXPRESSION = FaceExpression.LOOKING;
+    private static final int KEY_NEUTRAL = -1;
+
+    private final FaceTracker.FaceListener faceListener;
+    private final KeyToFaceMappings mappings;
+
+    private int currentPress = KEY_NEUTRAL;
+
+    public KeyboardFaceSource(FaceTracker.FaceListener faceListener, KeyToFaceMappings mappings) {
+        this.faceListener = faceListener;
+        this.mappings = mappings;
+    }
+
+    public boolean onKeyDown(int keyCode) {
+        if (sameKeyCodeIsTriggered(keyCode) || isStillUnsupportedCode(keyCode)) {
+            return false;
+        }
+        FaceExpression faceExpression = mappings.getFace(keyCode);
+        if (faceExpression != null) {
```

Comment:
Mmm this should never happen, as you are checking before if the mapping contains that keycode, and returning earlier in that case